### ISSUE TITLE
Bug Report: Download via non-ajax requests fail/FISH-7644

### ISSUE THREAD

**Issue by Lars5678** [payara/Payara#6340]
### Brief Summary

After migrating from Jakarta EE 8 (Payara 5.x) to Jakarta EE 10 (Payara 6.2023.4) we have problems with file download via Primefaces. The problem occurs as soon as the server is behind a proxy. The problem does not occur locally.

The problem has already been discussed in several forums:
https://forum.primefaces.org/viewtopic.php?t=80045
https://github.com/primefaces/primefaces/issues/9702
https://github.com/eclipse-ee4j/mojarra/issues/5202

The problem was localized in Mojarra 4.0 and fixed with [4.0.2](https://github.com/eclipse-ee4j/mojarra/releases/tag/4.0.2-RELEASE) (https://github.com/eclipse-ee4j/mojarra/issues/5165 / https://github.com/eclipse-ee4j/mojarra/pull/5166) .

We were able to solve the problem by setting the parameter `useBundledJsf=true` via `payara-web.xml` and delivering Mojarra 4.0.2 ourselves.

It would be great if you integrated Mojarra 4.0.2 in Payara in a timely manner.

### Expected Outcome

Payara supports and delivers Mojarra 4.0.2

### Current Outcome

Download via non-ajax requests fail because an attribute is missing in class impl/src/main/java/com/sun/faces/context/ExternalContextImpl.java

### Reproducer

I think a reproducer is not necessary.

If you need a reproducer please let me know.

However, this would then have to run behind a proxy in order to represent the problem.

### Operating System

Linux

### JDK Version

Amazon Corretto 17

### Payara Distribution

Payara Micro

**Comment by LarsBrenker** [comment]
Hi @felixif , I can understand your objection regarding the reproducer. I have always delivered a reproducer so far. In my opinion, however, the effort here is not in proportion to the benefit, since in my opinion it makes sense to import the Mojarra patch anyway.

Thank you for forwarding it to development.

**Comment by felixif** [comment]
Hi @LarsBrenker,

I do not have any objections when it comes to you not providing the reproducer. I appreciate your analysis, I know you have been contributing with issue reports for a while, and we really appreciate your input on the product. I apologise if it came across as an objection.

**Comment by LarsBrenker** [comment]
No, everything is OK. I didn't take it as an objection. I just wanted to say that I think it's good to create reproducers in general. We look forward to the hot fix.

**Comment by StrongKeyAbhishek** [comment]
This issues is still reproducible in 6.2023.10, is there any update on when this fix will be released with the updated Mojarra?

**Comment by felixif** [comment]
@StrongKeyAbhishek, I have verified the issue, but sadly it still sits in the backlog. 
I updated the internal issue, stating that it is still reproducible in Payara 6.2023.10 and 6.7.0.  I will discuss this issue in the next team meeting, and see if we can prioritize it. Thank you very much for updating this issue and bringing attention to it.

**Comment by tuerker** [comment]
This issue still persists in Payara 6.2024.3 in May 2024. 

Would be great to fix this problem asap. Many thanks in advance

**Comment by lprimak** [comment]
Yet another reason to move to updated Mojrarra
